### PR TITLE
boards: nucleo_g474re: Use HSE instead of HSI

### PR DESCRIPTION
**Pull request by alexanderwachter**
Change the board defconfig to use the provided 24MHz Crystal instead
of the internal resonator.
This improves the clock precision and stability.

### CODE REVIEW

File: boards/arm/nucleo_g474re/nucleo_g474re_defconfig [hunk 1]
```diff
@@ -27,16 +27,17 @@ CONFIG_CLOCK_CONTROL=y
 # Use PLLCLK for SYSCLK
 CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL=y
 
-# Use HSI (16MHz) to feed into PLL
-CONFIG_CLOCK_STM32_PLL_SRC_HSI=y
+# Use HSE (24MHz) to feed into PLL
+CONFIG_CLOCK_STM32_PLL_SRC_HSE=y
+CONFIG_CLOCK_STM32_HSE_CLOCK=24000000
 
 CONFIG_CLOCK_STM32_PLL_M_DIVISOR=4
```
Comment (FRASTM) [inline boards/arm/nucleo_g474re/nucleo_g474re_defconfig:34]: Proposal for an aditionnal comment to produce the PLLCLK at 170MHz, the combination would be
`CONFIG_CLOCK_STM32_PLL_M_DIVISOR=6`
with a 
`CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER=85`

**Comment by FRASTM** [comment]
This change could also apply to nucleo_g431rb board as well

File: boards/arm/nucleo_g474re/nucleo_g474re_defconfig (hunk 1 above)
Comment (alexanderwachter) [reply boards/arm/nucleo_g474re/nucleo_g474re_defconfig:34]: Done.

**Comment by FRASTM** [comment]
Frequency is set correctly on nucleo_g431rb with HSE for PLLCLK at 150MHz

**Review by FRASTM — APPROVED** [review]

**Comment by erwango** [comment]
@alexanderwachter, since it was in first, I propose you rebase on https://github.com/zephyrproject-rtos/zephyr/pull/24157, then we'll merge this one

**Comment by erwango** [comment]
@alexanderwachter can you rebase ?

**Comment by alexanderwachter** [comment]
@FRASTM The q output from the PLL should actually generate 48MHz right? Otherwise it is useless for USB.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**